ceph/ceph.py

```diff
@@ -1253,13 +1254,22 @@ def __connect(self):
         end_time = datetime.datetime.now() + self.outage_timeout
         while end_time > datetime.datetime.now():
             try:
-                self.__client.connect(
-                    self.ip_address,
-                    username=self.username,
-                    password=self.password,
-                    look_for_keys=self.look_for_keys,
-                    allow_agent=False,
-                )
+                if self.look_for_keys:
+                    self.__client.connect(
+                        self.ip_address,
+                        username=self.username,
+                        look_for_keys=self.look_for_keys,
+                        allow_agent=False,
+                        key_file_name=self.__private_key_file_path
```

Comment:
By default `__private_key_file_path` set to `None`. In this case don't need this if.. else block. first block should work.

Reply:
I have removed the password argument as well in the first block.

Reply:
Updated the changes, works without passing a password as well (when just ssh key is passed)